File: backend/models/Site.js
const mongoose = require('mongoose');

const SiteSchema = new mongoose.Schema({
  id: { type: String, required: true, unique: true },
  name: { type: String, required: true },
  location: { type: String, required: true },
  activeWorkers: { type: Number, default: 0 },
  latitude: { type: Number, required: true },
  longitude: { type: Number, required: true },
  geofenceRadius: { type: Number, default: 200 },
  clientName: String,
  clientGstin: String,
  clientEmail: String,
  clientContact: String,
  workOrderNo: String,
  workOrderDate: String,
  workOrderEndDate: String,
  billingRate: Number,
  companyName: String,
  username: { type: String, unique: true, sparse: true }, // Username for supervisor login
  password: { type: String, default: 'ambe123' }, // Password for supervisor login
  status: { type: String, enum: ['Active', 'Pending', 'Deleted'], default: 'Active' }
}, { timestamps: true });

module.exports = mongoose.model('Site', SiteSchema);

File: supervisor-android/app/src/main/java/com/ambe/supervisor/models/Site.java
package com.ambe.supervisor.models;

public class Site {
    private String id;
    private String name;
    private String location;
    private double latitude;
    private double longitude;
    private int geofenceRadius;

    public Site(String id, String name, String location, double latitude, double longitude, int geofenceRadius) {
        this.id = id;
        this.name = name;
        this.location = location;
        this.latitude = latitude;
        this.longitude = longitude;
        this.geofenceRadius = geofenceRadius;
    }

    public String getId() { return id; }
    public String getName() { return name; }
    public String getLocation() { return location; }
    public double getLatitude() { return latitude; }
    public double getLongitude() { return longitude; }
    public int getGeofenceRadius() { return geofenceRadius; }
}
